spooky-oysters/Bangazon-Command-Line-Application
# Kb update product menu

## Link to Ticket
Closes #8  User should be able to update product

## Description of Proposed Changes
- Product fields can be updated via CLI

## Steps to Test

Outline the steps to test

```sh
git fetch --all
git checkout kb-updateProductMenu
cd bangazon-cli.Tests
dotnet restore
dotnet test

..
cd bangazon-cli
dotnet run

```
1. Select 2 to activate a customer (create a customer, if needed)
1. Select 4 to add a product to active customer - DON'T add to shopping cart yet
1. Select 5 update product. Select product you would like to update. Follow prompts to update product name, price, description, quantity (for quantity, you should get the following prompt: "Update product quantity. Ex: 45 to add quantity. -45 to remove quantity.")
1. Check database to make sure product was updated correctly
1. From main menu, select 7 and select product to add to shopping cart
1. Select 5 to update product. Select product you would like to update. Select 4 quantity to update. You should get the following prompt: "Add additional quantity to product"
1. Check database to make sure product was updated correctly

## Impacted Areas in Application

List general components of the application that this PR will affect:

* UpdateProductMenu.cs
* ProductManager.cs - update method changed (no longer separate update methods for each Product property)
* ProductManager_Should.cs - test to update product fields refactored.

## Mentions @username

Tag users that need to review this code

## Definition of Done

1. The project must be fully documented. This includes the following:
    1. Complete README that documents the steps to install the code, how to install any dependencies, or system configuration needed.
    1. Every class must be documented with purpose, author, and methods.
    1. Every method must be documented with purpose and argument list - which itself must contain a short purpose for each argument.
1. The project must be able to run fully, and without errors, on each teammate's system.
1. Fulfills every requirement.
Every line of code has been peer reviewed.
1. For projects that require unit testing, core functionality must be identified and have at least one test for each.
